stdlib/builtin/thread.rbs

```diff
@@ -754,15 +755,15 @@ class Thread < Object
   #   # all exceptions inherited from Exception are prohibited.
   # }
   # ```
-  def self.handle_interrupt: (untyped hash) -> untyped
+  def self.handle_interrupt: [X] (Hash[untyped, Symbol] hash) { (nil arg) -> X } -> X
```

Comment:
It seems `Hash[Exception, Symbol]` means `instance_of?(Exception)`, not `kind_of?`.
Is there any way to declare?

Reply:
`Hash[singleton(Exception), :immediate | :on_blocking | :never]` ???